docs/asset-store/docs/06-02-bucket.md

```diff
@@ -0,0 +1,47 @@
+---
+title: Bucket
+type: Custom Resource
+---
+
+The `buckets.assetstore.kyma-project.io` CustomResourceDefinition (CRD) is a detailed description of the kind of data and the format used to define the name of the cloud storage bucket for storing assets. To get the up-to-date CRD and show the output in the `yaml` format, run this command:
+
+```
+kubectl get crd buckets.assetstore.kyma-project.io -o yaml
+```
+
+## Sample custom resource
+
+This is a sample resource that defines the storage bucket configuration.
+
+```
+apiVersion: assetstore.kyma-project.io/v1alpha1
+kind: Bucket
+metadata:
+  name: my-bucket
+  namespace: default
+```
+
+## Custom resource parameters
+
+This table lists all the possible parameters of a given resource together with their descriptions:
+
+
+| Parameter   |      Mandatory      |  Description |
+|:----------:|:-------------:|:------|
+| **metadata.name** |    **YES**   | Specifies the name of the CR. |
+| **metadata.namespace** |    **NO**   | Specifies the Namespace in which the CR is available. |
```

Comment:
please talk to @pkosiec there are already 2 spec fields implemented, not like we assumed at the beginning, spec.policy and spec.region

Reply:
Yes, there are two additional fields in `spec`:

1. `location` - location of the bucket, where it should be created. Here are listed all valid locations: https://docs.minio.io/docs/golang-client-api-reference.html#MakeBucket
If the field is empty, it means the default location will be used. So yeah, this is optional field.

Maybe I could make an enum with these values...

2. `policy` - here the things are more complicated. This is optional field as well. Default policy is empty, which means there is no policy for the bucket. If an user wants to change the policy, (s)he needs to provide policy in string:

something like that:

`{"Version": "2012-10-17","Statement": [{"Action": ["s3:GetObject"],"Effect": "Allow","Principal": {"AWS": ["*"]},"Resource": ["arn:aws:s3:::my-bucketname/*"],"Sid": ""}]}`

To be honest I need to dig into this topic, because I'm not able to provide more pieces of information about the whole `policy` topic. Things had been much simpler a year ago - the API has been changed recently.

Reply:
This is sample of the `Bucket` CR:

```
apiVersion: assetstore.kyma-project.io/v1alpha1
kind: Bucket
metadata:
  labels:
    controller-tools.k8s.io: "1.0"
  name: test-sample
  namespace: default
spec:
  region: "us-east-1"
  policy: >
    {
      "Version":"2012-10-17",
      "Statement":[
      {
        "Effect":"Allow",
        "Principal":{
          "AWS":[
            "*"
          ]
        },
        "Action":[
          "s3:GetBucketLocation",
          "s3:ListBucket"
        ],
        "Resource":[
          "arn:aws:s3:::ns-default-test-sample"
        ]
      },
      {
        "Effect":"Allow",
        "Principal":{
          "AWS":[
            "*"
          ]
        },
        "Action":[
          "s3:GetObject"
        ],
        "Resource":[
          "arn:aws:s3:::ns-default-test-sample/*"
        ]
      }
      ]
    }
```